### PR TITLE
Make partition keys required for container creates

### PR DESCRIPTION
**Pull request by johanste**

### CODE REVIEW

**Comment by binderjoe** [comment]
LGTM

File: azure/cosmos/__init__.py [hunk 1]
```diff
@@ -405,6 +406,7 @@ def create_container(
         :param session_token: Token for use with Session consistency.
         :param access_condition: Conditions Associated with the request.
         :param populate_query_metrics: Enable returning query metrics in response headers.
+        :param offer_throughput: The provisioned throughput for this offer.
```
Comment (mmacy) [inline azure/cosmos/__init__.py:409]: This comment doesn't apply *only* to this param, and in this case likely isn't in your wheelhouse, but leaving this here to record our **need for better descriptions of parameters** in general.

---
As a developer I would first ask "What is 'throughput' in this context? What does this `int` represent?" Followed closely would be "Are there minimum and maximum values for this parameter? Any other restrictions?"

Recommend that you specify that this value is the number of **request units (RU)**, and include any other limits or requirements for this parameter. Also consider including a link to the conceptual doc [Request units in Azure Cosmos DB](https://docs.microsoft.com/azure/cosmos-db/request-units):

`https://docs.microsoft.com/azure/cosmos-db/request-units`

File: examples/examples.py [hunk 2]
```diff
@@ -62,51 +64,50 @@
...
+    container.upsert_item(
+        dict(id=f"item{i}", productName="Widget", productModel=f"Model {i}")
     )
 # [END upsert_items]
 
 # Modify an existing item in the container
 # [START update_item]
-item = container.get_item('item2')
-item['productModel'] = 'DISCONTINUED'
+item = container.get_item("item2", partition_key="Widget")
```
Comment (mmacy) [inline examples/examples.py:88]: This line was throwing intermittently; I've opened #19 to ensure we insert the items into the correct container in the upsert above.

**Review by mmacy — COMMENTED** [review]
I'm late to the game on this, but wanted to explicitly call out our need for better parameter descriptions across the board.

File: azure/cosmos/__init__.py (hunk 1 above)
Comment (johanste) [reply azure/cosmos/__init__.py:409]: Yeah, I'll happily admit that, for the most part, I have just done a lift-and-shift of the existing documentation for parameters. So, yeah, we should improve this.

**Pull request merged**